stunnel/stunnel.ml

```diff
@@ -155,16 +164,20 @@ let config_file verify_cert extended_diagnosis host port legacy =
       Some x -> Printf.sprintf "TIMEOUTidle = %d" x)
     ; Printf.sprintf "connect=%s:%d" host port
     ]
-  ; if is_fips then [] else ["fips=no"]
-  ; if extended_diagnosis then ["debug=4"] else []
+  ; if is_fips then ["fips=yes"] else ["fips=no"]
+  ; if extended_diagnosis then ["debug=authpriv.7"] else ["debug=authpriv.5"]
   ; if verify_cert then
       ["verify=2"
+      ; sprintf "checkHost=%s" host
       ; sprintf "CAfile=%s" certificates_bundle_path
-      ; sprintf "CRLpath=%s" crl_path
+      ; (match is_valid_crl_path crl_path with
+         | true -> sprintf "CRLpath=%s" crl_path
+         | false -> "")
       ]
     else []
   ; if legacy then
-      [ "sslVersion = all"
+      [ "sslVersionMax = TLSv1.2"
+      ; "sslVersionMin = TLSv1"
```

Comment:
Note that as part of the other REQ this code will get dropped completely. For now I think it is good.